web/components/nav/group-nav-bar.tsx

```diff
@@ -72,23 +76,23 @@ function NavBarItem(props: {
   currentPage: string
   onClick: (key: string) => void
 }) {
-  const { item, currentPage } = props
+  const { item, currentPage, onClick } = props
   const track = trackCallback(
     `group navbar: ${item.trackingEventName ?? item.name}`
   )
 
   return (
-    <button onClick={() => props.onClick(item.key ?? '#')}>
+    <Link href={item.href ?? '#'}>
       <a
         className={clsx(
           'block w-full py-1 px-3 text-center hover:bg-indigo-200 hover:text-indigo-700',
           currentPage === item.key && 'bg-gray-200 text-indigo-700'
         )}
-        onClick={track}
+        onClick={() => (onClick?.(item.key ?? '#'), track())}
```

Comment:
imo using an onClick handler that calls Next router is an anti-pattern, and instead you should just put the correct href (like `groupSlug/market`)